packages/core/src/new-api/internal/deployer.ts

```diff
@@ -56,7 +56,11 @@ export class Deployer {
   ): Promise<DeploymentResult> {
     const module = this._moduleConstructor.construct(moduleDefinition);
 
-    await validate(module, this._artifactResolver);
+    await validate(
+      module,
+      this._artifactResolver,
+      deploymentParameters[module.id]
```

Comment:
Wouldn't passing just this module's params limit the ability to validate submodules?

Reply:
Oh I just noticed that we aren't validating submodules at all.

Should this be address in this PR or a new one? /cc @kanej

Reply:
fixed in 852f58f